applications/nrf5340_audio/src/audio_module/src/ablk_api.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright(c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _ABLK_API_H_
+#define _ABLK_API_H_
+
+#include <zephyr/kernel.h>
+
+/**
+ * @brief Value to define the data type carried by the block.
+ */
+enum ablk_type {
+	/* The audio data block type is undefined */
+	ABLK_TYPE_UNDEFINED = 0,
+
+	/* The audio data block is raw PCM */
+	ABLK_TYPE_PCM,
+
+	/* The audio data block is coded in LC3 */
+	ABLK_TYPE_LC3
+};
+
+/**
+ * @brief Value to define the PCM interleaving type.
+ */
+enum ablk_interleaved {
+	/* The interleaving is undefined */
+	ABLK_INTERLEAVE_UNDEFINED = 0,
+
+	/* Audio samples are deinterleaved within the buffer */
+	ABLK_DEINTERLEAVED,
+
+	/* Audio samples are interleaved within the buffer */
+	ABLK_INTERLEAVED
+};
+
+/**
+ * @brief A structure describing the PCM data format
+ */
+struct ablk_pcm_format {
+	/* The PCM sample rate in Hz */
+	uint32_t sample_rate_Hz;
+
+	/* Number of valid bits for a sample */
+	uint8_t bits_per_sample;
+
+	/* Number of bits used to carry a sample of size bits_per_sample */
+	uint8_t carrier_size;
+
+	/* A flag indicating if the PCM block is sample interleaved or not */
+	enum ablk_interleaved interleaved;
+
+	/* Number of channels in the PCM block */
+	uint8_t number_channels;
+
+	/* A 32 bit mask indicating which channel(s) are active within
+	 * the PCM block. A bit set indicates the channel is active.
+	 * Note: This will follow the ANSI/CTA-861-G’s Table 34 codes
+	 * for speaker placement (as used by Bluetooth Low Energy Audio).
+	 */
+	uint32_t channel_map;
+};
+
+/**
+ * @brief A structure giving the description of the audio block
+ */
+struct ablk_block {
+	/* Indicates the data type of the block */
+	enum ablk_type data_type;
+
+	/* A pointer to the raw or coded data (e.g., PCM, LC3, etc.) */
+	uint8_t *data;
+
+	/* The size in bytes of the raw data block */
```

Comment:
Nitpick: The comment is not quite right, as the data block may be raw or encoded. 
```suggestion
	/* The size in bytes of the data */
```